gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreUtils.java

```diff
@@ -256,6 +256,23 @@ public static SerDeInfo getSerDeInfo(HiveRegistrationUnit unit) {
     return si;
   }
 
+  /**
+   * Util for detecting if a table has a complex union (aka non-optional unions) column types. Iceberg currently does
+   * not natively support uniontype because types differently as structs. Until this feature is added in iceberg, this
+   * function can be used to detect these types of tables for gobblin specific workarounds
+   *
+   * See https://github.com/apache/iceberg/issues/189
+   *
+   * @param t
+   * @return if hive table contains complex uniontype columns
+   */
+  public static boolean containsUnionTypeColumn(HiveTable t) {
```

Comment:
Can we have another method that takes Hive native table object instead of Gobblin Hive Table wrapper object and one calling the other?